gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java

```diff
@@ -224,18 +238,33 @@ private Map<String, BoundField> getBoundFields(Gson context, TypeToken<?> type,
         }
         // The accessor method is only used for records. If the type is a record, we will read out values
         // via its accessor method instead of via reflection. This way we will bypass the accessible restrictions
-        // If there is a static field on a record, there will not be an accessor. Instead we will use the default
-        // field logic for dealing with statics.
         Method accessor = null;
-        if (isRecord && !Modifier.isStatic(field.getModifiers())) {
-          accessor = ReflectionHelper.getAccessor(raw, field);
+        if (isRecord) {
+          // If there is a static field on a record, there will not be an accessor. Instead we will use the default
+          // field logic for dealing with statics. For deserialization the field is ignored.
```

Comment:
Just to clarify: by default, static fields in records will be neither serialized nor deserialized, right? Only if that weird `excludeFieldsWithModifiers` method is called? Because I definitely don't think static fields should be serialized.

Reply:
Yes that is correct, see default excluded modifiers here:
https://github.com/google/gson/blob/c2458bf876ef52a4489af937f7dd7c2709b74fb9/gson/src/main/java/com/google/gson/internal/Excluder.java#L55

I have mainly added this to handle it in a reasonable way instead of failing with some obscure exception.

For static fields maybe there are legit use cases, e.g. to automatically encode a version number:
```java
record Response(
   ...
) {
    public static final int VERSION = 1;
}
```
Though I am not sure if someone is actually using this, and it is certainly not obvious (and probably not even officially supported) to use `excludeFieldsWithModifiers` for this.

If you want I can adjust it to also always exclude static fields on serialization, and we can then wait for someone to request this feature.

Reply:
Yes, I think by default static fields should not be serialized. I suppose they should be if someone calls `excludeFieldsWithModifiers(0)`, though even then I don't really understand why people would want that. I'm not really convinced by the `VERSION` use case. We would serialize that field, and then what would we do on deserialization? Do we check that the version matches? Do we try to overwrite the `VERSION` field? If we don't, isn't that inconsistent? I think if you really want a version then it should either be an instance field or you should use a custom `TypeAdapter`.

(I looked at the history, and apparently `excludeFieldsWithModifiers` was already present in the first version on GitHub, dated 2008. So who knows why it was added. For what it's worth, Google's code has only a handful of calls to this method, and exactly one that is including `STATIC`, perhaps accidentally.)

Reply:
> I'm not really convinced by the `VERSION` use case. We would serialize that field, and then what would we do on deserialization?

It would probably be most useful for services which only serialize data, e.g. a REST API which generates the response. At least that is the most reasonable situation I can come up with at the moment.

It appears the `excludeFieldsWithModifiers` trick is known, see [this StackOverflow answer](https://stackoverflow.com/a/33666187).